core/src/ops/nn/reduce.rs

```diff
@@ -142,44 +143,58 @@ impl Reducer {
     // tricky (not associative)
     unsafe fn sum<T>(&self, axes: &[usize], input: &Tensor) -> Tensor
     where
-        T: Copy + Datum + num_traits::Zero,
+        T: Copy + Datum + num_traits::Zero + Sum,
+        f16: AsPrimitive<T>,
+        f32: AsPrimitive<T>,
     {
         if axes.len() == 0 {
             return input.to_owned();
         }
+
         let mut output: Option<ArrayD<T>> = None;
-        for axis in axes.iter() {
-            let current_input = output
+        for axis in axes.iter().cloned() {
```

Comment:
Nitpicking: can we dispense with the cloned() ? These are just usize, right ? so `&axis` or at least `.copied()` should be enough?

Reply:
Yes, I just didn't want to dereference axis everywhere in the scope. I'll change that to `.copied()`.